inmem.go

```diff
@@ -120,6 +121,27 @@ func (a *AggregateSample) String() string {
 	}
 }
 
+// NewInmemSinkFromURL creates an InmemSink from a URL. It is used
+// (and tested) from NewMetricSinkFromURL.
+func NewInmemSinkFromURL(u *url.URL) (MetricSink, error) {
+	params := u.Query()
+	if len(params["interval"]) < 1 {
```

Comment:
You could also collapse this with the `ParseDuration` call and just blindly try to parse whatever `Values.Get()` gives you. That way if it was empty you would just get an error like `invalid interval: ""`.